standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/retryhandling/RetryCallProperties.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore.txn.retryhandling;
+
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.datanucleus.transaction.TransactionIsolation;
+
+import java.util.function.Function;
+
+/**
+ * Contains all the properties which can alter the behavior of a retry-call in 
+ * {@link org.apache.hadoop.hive.metastore.txn.retryhandling.RetryHandler}.
+ */
+public class RetryCallProperties {
+
+  private String dataSource;
+  private String caller = null;
+  private boolean retryOnDuplicateKey = false;
+  private int transactionIsolationLevel = TransactionIsolation.READ_COMMITTED;
```

Comment:
why do we keep the txn isolation level here? txn propagation is set to Required in RetryHandler method directly?

Reply:
Added propagation level too. I think better to have control over these, even if currently all methods using the default value.